# U.S. Medical Insurance Costs

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('insurance.csv')
df.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [3]:
# explore the data info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


We found that this data contains 1338 rows with 7 cols.
There's no null value in this dataset.

In [4]:
# explore the decriptive statistic of the dataset

df.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

For ages data: Average age is 39.20 years, maximum age is 64 years and minimum age is 18 years.
For bmis data: Average bmi is 30.66, maximum bmi is 53.13 and minimum bmi is 15.96
For charges data: Average charges is 13270 dollars, maximum charges is 63770.43 dollars and minimum charges is 1121.87 dollars

In [5]:
# Now, let's calculate the mean of bmi, age and charges of this dataset

avg_bmi = df['bmi'].mean()
avg_age = df['age'].mean()
avg_charge = df['charges'].mean()

Let's take a look into the data in each column

I will analyze the data in each column to learn more about the patience information.

In [6]:
# Q1. How many male and female?

df['sex'].value_counts()

male      676
female    662
Name: sex, dtype: int64

In [7]:
# Q2. How many number of the patience live in each region?

df['region'].value_counts()

southeast    364
northwest    325
southwest    325
northeast    324
Name: region, dtype: int64

In [8]:
# Q3. How many patience are smoker and non-smoker?

df['smoker'].value_counts()

no     1064
yes     274
Name: smoker, dtype: int64

In [21]:
# Q4. How many smoker are male and female?
df[df['smoker'] == 'yes']['sex'].value_counts()

male      159
female    115
Name: sex, dtype: int64

In [29]:
# Q5. What about the data compare between male and female?
format_int = (lambda x: x.format('%.2f'))

avg_male_age = df[df['sex'] == 'male']['age'].mean()
avg_male_bmi = df[df['sex'] == 'male']['bmi'].mean()
avg_male_charge = df[df['sex'] == 'male']['charges'].mean()

avg_female_age = df[df['sex'] == 'female']['age'].mean()
avg_female_bmi = df[df['sex'] == 'female']['bmi'].mean()
avg_female_charge = df[df['sex'] == 'female']['charges'].mean()

max_male_age = df[df['sex'] == 'male']['age'].max()
max_male_bmi = df[df['sex'] == 'male']['bmi'].max()
max_male_charge = df[df['sex'] == 'male']['charges'].max()

min_male_age = df[df['sex'] == 'male']['age'].min()
min_male_bmi = df[df['sex'] == 'male']['bmi'].min()
min_male_charge = df[df['sex'] == 'male']['charges'].min()

max_female_age = df[df['sex'] == 'female']['age'].max()
max_female_bmi = df[df['sex'] == 'female']['bmi'].max()
max_female_charge = df[df['sex'] == 'female']['charges'].max()

min_female_age = df[df['sex'] == 'female']['age'].min()
min_female_bmi = df[df['sex'] == 'female']['bmi'].min()
min_female_charge = df[df['sex'] == 'female']['charges'].min()

print(f'The average age of all patience is {avg_age} years. The average age of male is {avg_male_age} years and female is {avg_female_age} years.')
print(f'The average bmi of all patience is {avg_bmi}. The average bmi of male is {avg_male_bmi} and female is {avg_female_bmi}')
print(f'The average charges of all patience is {avg_charge} dollars. The average charges of male is {avg_male_charge} dollars and female is {avg_female_charge} dollars')

The average age of all patience is 39.20702541106129 years. The average age of male is 38.917159763313606 years and female is 39.503021148036254 years.
The average bmi of all patience is 30.663396860986538. The average bmi of male is 30.943128698224832 and female is 30.377749244713023
The average charges of all patience is 13270.422265141257 dollars. The average charges of male is 13956.751177721886 dollars and female is 12569.57884383534 dollars


In [32]:
# Q6. The average charges of smoker and non-smoker

df.groupby('smoker')['charges'].mean()

smoker
no      8434.268298
yes    32050.231832
Name: charges, dtype: float64

So, the patience who smoker will pay a big amount of charges. It's higher than the overall average charges.

In [37]:
# Q7. Want to know about the patience information who pay the highest charges

df[df['charges'] == df['charges'].max()]

age     sex    bmi  children smoker     region      charges
543   54  female  47.41         0    yes  southeast  63770.42801

In [38]:
# Q8. Want to know about the patience information who pay the lowest charges

df[df['charges'] == df['charges'].min()]

age   sex    bmi  children smoker     region    charges
940   18  male  23.21         0     no  southeast  1121.8739

If you see the data for both the highest and the least charges
It's shown that the female who pay highest charges, smoke and is the eldery person with the bmi higher than average bmis
when compare with the lowest charges: a teenage boy who not smoke and the bmi is lower than the average bmi.

In [39]:
df.corr()

age       bmi  children   charges
age       1.000000  0.109272  0.042469  0.299008
bmi       0.109272  1.000000  0.012759  0.198341
children  0.042469  0.012759  1.000000  0.067998
charges   0.299008  0.198341  0.067998  1.000000

And if you see the correlation table: A strong positive correlation between charges and age. As well as between charges and bmi.
A slight positive correlation between children and charges.

In [40]:
# Q9. The average charges for smoker patience, groupby sex

df[df['smoker'] == 'yes'].groupby('sex')['charges'].mean()

sex
female    30678.996276
male      33042.005975
Name: charges, dtype: float64

In [41]:
# Q10. The average charges for non-smoker patience, groupby sex

df[df['smoker'] == 'no'].groupby('sex')['charges'].mean()

sex
female    8762.297300
male      8087.204731
Name: charges, dtype: float64